rosidl_generator_cpp/resource/action__traits.hpp.em

```diff
@@ -0,0 +1,85 @@
+@# Included from rosidl_generator_cpp/resource/idl__traits.hpp.em
```

Comment:
Does the `resource/idl__traits.hpp.em` template need to be updated in order to include this new file?

Also the test doesn't cover actions yet which based on the above seems to be broken.

Reply:
It does.

Also, because action generation depends on `action_msgs`, it's likely going to have to reside in a higher package.  I discussed with @jacobperron, and I'm going to move forward with putting the trait tests in `rclcpp_action`, unless there is another alternative.

Reply:
We can't build test action interfaces in `rosidl_generator_cpp` because of a circular dependency on `action_msgs`. So unless we can test `is_action` without generating an action, then until we remove the circular dependency we won't be able to add tests to this package.